Adds stealthy Bikehorn sword to traitor Clown

Wilchenx opened the pull request:
A clown can now buy an esword disguised as a bikehorn through his uplink for 10 tc.

It actives through alt-clicking.

I know this is a straight buff to normal eswords because of the hiding factor, but fuck that only clowns have it.

🆑
rscadd: adds a new clown traitor item: A bikehorn esword. Activate it by alt-clicking.
/🆑

https://user-images.githubusercontent.com/7501474/29424708-0ea53cfe-8381-11e7-84c2-1d162d528868.png

original code by Time-Green

Commit:
Merge pull request #1 from tgstation/master

sad

Commit:
Merge pull request #2 from tgstation/master

asda

Commit:
Update uplink_item.dm

Commit:
Update energy.dm

Commit:
Update energy.dm

Cyberboss commented on code/game/objects/items/melee/energy.dm:
```diff
@@ -228,3 +228,62 @@
+	var/obj/item/device/assembly/bikehorn/horn = new()
+	
+	/obj/item/weapon/melee/energy/sword/bikehorn/New()
+	..()
```
Initialize and . = ..()

Cyberboss commented on code/game/objects/items/melee/energy.dm:
```diff
@@ -228,3 +228,62 @@
+	..()
+	name = horn.name
+	icon_state = horn.icon_state
+	icon = horn.icon
```
use update_icon

Cyberboss commented on code/game/objects/items/melee/energy.dm:
```diff
@@ -228,3 +228,62 @@
+	name = ""
+	desc = "An energy blade extends from the bikehorn. Where's your god now?"
+	var/canSword = FALSE
+	var/obj/item/device/assembly/bikehorn/horn = new()
```
don't create the horn here

Commit:
fuck

Cyberboss commented on code/game/objects/items/melee/energy.dm:
```diff
@@ -228,3 +228,62 @@
+	name = horn.name
+	icon_state = horn.icon_state
+	icon = horn.icon
+	horn.forceMove(src)
```
create the horn here with src as the loc and remove this line

Cyberboss commented on code/game/objects/items/melee/energy.dm:
```diff
@@ -228,3 +228,62 @@
+ /obj/item/weapon/melee/transforming/energy/sword/bikehorn/attack_self(mob/living/carbon/L)
+	if(canSword)
+		activate(L)
+		playsound(loc, 'sound/items/bikehorn.ogg', 80, 1)
```
playsound(src, ...

Cyberboss commented on code/game/objects/items/melee/energy.dm:
```diff
@@ -228,3 +228,62 @@
+		if(canSword)
+			L << "<span class='notice'>You squeeze the [name]. Honking it will now extend an esword.</span>"
+		else
+ 			L << "<span class='notice'>You squeeze the [name]. It functions like a normal bikehorn again.</span>"
```
to_chat and [src]

Cyberboss commented on code/game/objects/items/melee/energy.dm:
```diff
@@ -228,3 +228,62 @@
+	if(L.mind.special_role)
+		canSword = !canSword
+		if(canSword)
+			L << "<span class='notice'>You squeeze the [name]. Honking it will now extend an esword.</span>"
```
to_chat and [src]

Cyberboss commented on code/game/objects/items/melee/energy.dm:
```diff
@@ -228,3 +228,62 @@
+	if(!active)
+		horn.attack(M,user)
+	else
+		playsound(loc, 'sound/items/bikehorn.ogg', 80, 1)
```
playsound(src, ...

Cyberboss commented on code/game/objects/items/melee/energy.dm:
```diff
@@ -228,3 +228,62 @@
+	var/obj/item/device/assembly/bikehorn/horn = new()
+	
+/obj/item/weapon/melee/energy/sword/bikehorn/New()
+	..()
```
Initialize and . = ..()

Cyberboss commented on code/game/objects/items/melee/energy.dm:
```diff
@@ -228,3 +228,62 @@
+	else
+		..()
+ 
+ /obj/item/weapon/melee/transforming/energy/sword/bikehorn/setIcon()
```
use update_icon

Cyberboss commented on code/game/objects/items/melee/energy.dm:
```diff
@@ -228,3 +228,62 @@
+	..()
+	name = horn.name
+	icon_state = horn.icon_state
+	icon = horn.icon
```
use update_icon

Commit:
fixes a few things

if something is retarded blame kevinz (unironically, he helped me out with a few things)

The-clown-known-as-Pepe commented:
still needs work but i can't do it now

InsaneHyena commented:
>Downvoted by people with no sense of humour
Does it honk when you hit enemies instead of making esword noises? Please say yes.

DoctoreMad commented:
Its not that I don't have a sense of humor I just think banana peel triple-e-sword is a better meme.

Kraseo commented:
>a stealth bikehorn that costs the same as a regular esword for the le epic grief job

uhhhhhhhhhhhhh

Commit:
changes cost to 10

The-clown-known-as-Pepe commented:
@Kraseo changes to 10
@InsaneHyena yes it honks

Super3222 commented:
@The-clown-known-as-Pepe drag and drop images into the chat box / pr description to upload them.
also, the PR he's porting is: https://github.com/yogstation13/yogstation/pull/2483

MoreRobustThanYou commented:
Port the invisible spray / clowncart / superglue from /vg/ as clown items.

Supermichael777 commented:
your avatar image violates our standards

- National Socialist Party of Germany content, National Socialist Party of Germany related content, or National Socialist Party of Germany references

Kraseo commented:
Thanks, Superminimod777!

Qustinnus commented on code/game/objects/items/melee/energy.dm:
```diff
@@ -228,3 +228,65 @@
+	if(!active)
+		horn.Crossed(L)
+	else
+		..()
```
.=..()

Qustinnus commented on code/game/objects/items/melee/energy.dm:
```diff
@@ -228,3 +228,65 @@
 	icon_state = "lightblade"
 	item_state = "lightblade"
+
+/obj/item/weapon/melee/transforming/energy/sword/bikehorn
```
change all /obj/item/weapon/melee/transforming/energy/sword/bikehorn to /obj/item/melee/transforming/energy/sword/bikehorn

Qustinnus commented on code/game/objects/items/melee/energy.dm:
```diff
@@ -228,3 +228,65 @@
 	item_state = "lightblade"
+
+/obj/item/weapon/melee/transforming/energy/sword/bikehorn
+	name = ""
```
🤔

Qustinnus commented on code/modules/uplink/uplink_item.dm:
```diff
@@ -1248,6 +1248,13 @@ GLOBAL_LIST_EMPTY(uplink_items) // Global list so we only initialize this once.
 
+/datum/uplink_item/role_restricted/bikehornsword
+	name = "Energy Sword Bikehorn"
+	desc = "A bikehorn with a built-in energy sword. Activate through Alt-Click."
```
boring description

Time-Green replied:
No it's not, it's a good description

Chronitonity commented:
atleast make it semi interesting instead of adding just another esword but a bike horn, thats just lazy

The-clown-known-as-Pepe commented:
it honks when ignited and on hit, its super stealthy

kevinz000 commented on code/game/objects/items/melee/energy.dm:
```diff
@@ -228,3 +228,65 @@
+		icon_state = horn.icon_state
+ 
+/obj/item/weapon/melee/transforming/energy/sword/bikehorn/AltClick(mob/living/carbon/L)
+	if(L.mind.special_role)
```
i really don't like the idea of something as dangerous as an esword being able to be easily hidden without any way to detect it short of carrying research scanners.
it'll just encourage security to stuff everything in the deconner.

The-clown-known-as-Pepe replied:
we don't see that done with pie cannon do we?

WJohn replied:
the existence of bad items doesn't justify adding more bad items

Ratvar commented:
Make it honk 200% louder in active state, remove R&D check, allow only clowns to unflap blade. Everyone knows when it hits, everyone wants valids, nobody can check which clown did it. Change item to glued sharpended toy blade you attach to bike horn to create honksword?  Just so shitcurity cannot "Two Bikehorns in pocket policy here, prepare to die, you have 1 horn too much.".
Or add action button to convert any bikehorn/duck into sword instead of weapon anyone will check and take? Trade new sword from fellow cluwne in exchange of emag/bullet in the head.

DoctoreMad commented:
In english please

ChangelingRain commented on code/game/objects/items/melee/energy.dm:
```diff
@@ -228,3 +228,65 @@
+	name = ""
+	desc = "An energy blade extends from the bikehorn. Where's your god now?"
+	var/canSword = FALSE
+	var/obj/item/weapon/bikehorn = new()
```
don't do this, just make the horn in update_icon

Cheridan commented:
uninspired
not funny

jughu commented:
Sprites!

Shadowlight213 commented:
I'm gonna use this as justification to steal the clown's bikehorn every round as sec.

KingPhilipIII commented:
@Shadowlight213  As though people don't already slip the clown to steal all his stuff already.

Shadowlight213 commented:
I'll have more justification to abuse the clown

KingPhilipIII commented:
Fair enough.

YakumoChen commented:
port clown cart and bananarangs, not this

Cyberboss commented:
Yeah, not gonna do this
